src/main/kotlin/no/nav/k9/los/integrasjon/abac/PepClient.kt

```diff
@@ -268,50 +172,69 @@ class PepClient constructor(
-        val requestBuilder = when (oppgave.oppgavetype.eksternId) {
-            "k9sak" -> {
-                XacmlRequestBuilder()
-                    .addEnvironmentAttribute(ENVIRONMENT_PEP_ID, "srvk9los")
-                    .addResourceAttribute(RESOURCE_DOMENE, DOMENE)
-                    .addResourceAttribute(RESOURCE_TYPE, TILGANG_SAK)
-                    .addActionAttribute(ACTION_ID, "update")
-                    .addResourceAttribute("no.nav.abac.attributter.resource.k9.behandlings_uuid", oppgave.eksternId)  //TODO los skal ikke kjenne til denne detaljen. Oppgavetype må utvides med attributtreferanse
-                    .addAccessSubjectAttribute(SUBJECT_TYPE, INTERNBRUKER)
-                    .addAccessSubjectAttribute(SUBJECTID, bruker.brukerIdent!!)
-            }
-            "k9klage" -> {
-                XacmlRequestBuilder()
+    private suspend fun harTilgang(
+        oppgavetype: String,
+        action: Action,
+        saksnummer: String?,
+        aktorId: String?,
+        auditlogging: Boolean
+    ): Boolean {
+        val identTilInnloggetBruker = azureGraphService.hentIdentTilInnloggetBruker()
+        return when (oppgavetype) {
+            "k9sak", "k9klage" -> {
+                if (auditlogging) k9Auditlogger.loggTilgangK9Sak(saksnummer!!, aktorId!!, identTilInnloggetBruker, action)
+                evaluate(XacmlRequestBuilder()
                     .addEnvironmentAttribute(ENVIRONMENT_PEP_ID, "srvk9los")
                     .addResourceAttribute(RESOURCE_DOMENE, DOMENE)
                     .addResourceAttribute(RESOURCE_TYPE, TILGANG_SAK)
-                    .addActionAttribute(ACTION_ID, "create")
-                    .addResourceAttribute("no.nav.abac.attributter.resource.k9.saksnr", oppgave.hentVerdi("saksnummer")!!)  //TODO los skal ikke kjenne til denne detaljen. Oppgavetype må utvides med attributtreferanse
+                    .addActionAttribute(ACTION_ID, action)
                     .addAccessSubjectAttribute(SUBJECT_TYPE, INTERNBRUKER)
-                    .addAccessSubjectAttribute(SUBJECTID, bruker.brukerIdent!!)
+                    .addAccessSubjectAttribute(SUBJECTID, identTilInnloggetBruker)
+                    .addResourceAttribute(RESOURCE_SAKSNR, saksnummer!!)
+                )
             }
-            else -> throw NotImplementedError("Støtter kun tilgangsoppslag på k9klage og k9sak")
-        }
+            "k9punsj" -> {
+                // ABAC bør støtte tilgangssjekk på punsj-oppgaver med journalpostId, for å unngå to kall og logikk her
+
+                val tilgangForSøker =
+                    evaluate(
+                        XacmlRequestBuilder()
+                            .addEnvironmentAttribute(ENVIRONMENT_PEP_ID, "srvk9los")
+                            .addResourceAttribute(RESOURCE_DOMENE, DOMENE)
+                            .addResourceAttribute(RESOURCE_TYPE, TILGANG_SAK)
```

Comment:
Jeg tror ikke denne ressurstypen vil virke for aktører. Her bør det nok heller brukes TILGANG_SAK_KODE6 og TILGANG_SAK_KODE7OGEGENANSATT